internal/archiver/archiver.go

```diff
@@ -388,19 +389,33 @@ func (arch *Archiver) Save(ctx context.Context, snPath, target string, previous
 
 		// use previous list of blobs if the file hasn't changed
 		if previous != nil && !fileChanged(fi, previous, arch.IgnoreInode) {
-			debug.Log("%v hasn't changed, using old list of blobs", target)
-			arch.CompleteItem(snPath, previous, previous, ItemStats{}, time.Since(start))
-			arch.CompleteBlob(snPath, previous.Size)
-			fn.node, err = arch.nodeFromFileInfo(target, fi)
-			if err != nil {
-				return FutureNode{}, false, err
+			contentsOK := true
+			// check if all blobs are contained in index
+			for _, id := range previous.Content {
+				if !arch.Repo.Index().Has(id, restic.DataBlob) {
+					contentsOK = false
+				}
 			}
 
-			// copy list of blobs
-			fn.node.Content = previous.Content
+			if contentsOK {
+				debug.Log("%v hasn't changed, using old list of blobs", target)
+				arch.CompleteItem(snPath, previous, previous, ItemStats{}, time.Since(start))
+				arch.CompleteBlob(snPath, previous.Size)
+				fn.node, err = arch.nodeFromFileInfo(target, fi)
+				if err != nil {
+					return FutureNode{}, false, err
+				}
 
-			_ = file.Close()
-			return fn, false, nil
+				// copy list of blobs
+				fn.node.Content = previous.Content
+
+				_ = file.Close()
+				return fn, false, nil
+			} else {
+				debug.Log("%v hasn't changed, but contents are missing!", target)
+				// There are contents missing - inform user!
+				fmt.Fprintf(os.Stderr, "Warning: Contents of %v are missing in repository! They are added now..\n", target)
```

Comment:
They're not added yet, though. The save may still fail.

"Warning: contents of %v missing from index; storing fresh copy" (?)

Reply:
changed.